core/iwasm/fast-jit/fe/jit_emit_control.c

```diff
@@ -1037,6 +1093,12 @@ jit_compile_op_br_if(JitCompContext *cc, uint32 br_depth,
     JitInsn *insn, *insn_select = NULL, *insn_cmp = NULL;
     bool copy_arities;
 
+#if WASM_ENABLE_THREAD_MGR != 0
+    /* Insert suspend check point */
+    if (!jit_check_suspend_flags(cc))
+        return false;
+#endif
```

Comment:
Don't put these lines here, since below optimization may depend on the original last two insns.
Move them to be before L1095(old code, or L1157 for modified code) instead: before this line: `CREATE_BASIC_BLOCK(if_basic_block);`

Reply:
I tried to move it, but then it failed on three spec tests: fac br_if and unwind